### PR TITLE
Issue 1009 horrible prompt

### LINKED ISSUE #1009: setup_runtime messes with prompt!
commit b0d52306befd855bc333f7c2884448cbacd7f1b8 decided to change the users prompt. 

It should AT LEAST be introduced in the .csh script at well. These 2 MUST be kept in sync. We can't tweak one w/o the other

### PR DESCRIPTION
**Pull request by doutriaux1**
fix #1009 @aashish24  @dlonie let's approve this soon. Thanks

### CODE REVIEW

**Comment by dlonie** [comment]
I actually kinda like having the prompt, it makes it easier to tell whether uvcdat has been sourced or not -- it might be worth adding the version number as well for users that might have multiple versions laying around.

I agree with disabling it by default -- users can add the enable variable to their shell init if they want it (I personally don't like when things mess with my PS2 unexpectedly).

Can we rename the variable though? "IN_PROMPT" doesn't really match the feature -- how about "ENABLE_PROMPT"?

**Comment by doutriaux1** [comment]
ok will do, good suggestions @dlonie 
Actually what about the following UVCDAT_ENABLE_PROMPT_BEGINING and UVCDAT_ENABLE_PROMPT_END so you can decide where you want it?
Will also store the result in UVCDAT_PROMPT_STRING so you can actually put it anyway/where/color you want in your prompt.
Does that sound ok?

**Comment by doutriaux1** [comment]
and will stick version at least in the UVCDAT_PROMPT_STRING

**Comment by dlonie** [comment]
Git can do something similar in a very nice way. See:

https://github.com/git/git/blob/master/contrib/completion/git-prompt.sh

Basically, it defines some bash functions that can be inserted into the PS2 string that displays information like current branch, etc. I think this is the ideal way to achieve this -- rather than modifying PS2 ourselves, define some function/var that the user can add to their prompt, where they want it (in case they want more control than just beginning / end.

But I'm fine with either way, just wanted to point the above out.

**Comment by doutriaux1** [comment]
yes I use that and I love it, but I'm not looking forward spending a week to make a nice prompt :wink: so I'll just push a quick and dirty-ish way. IF someone care to make it even better, that would be awesome

**Comment by doutriaux1** [comment]
plus the one above is bash and we should (even though some PR have not done so) support csh as well.

**Comment by doutriaux1** [comment]
@dlonie ok what about this?

File: CMake/cdat_modules_extra/setup_runtime.csh.in [hunk 1]
```diff
@@ -1,6 +1,14 @@
 # Main install prefix set by user or post install script:
 # UVCDAT_INSTALL_PREFIX
 
+setenv UVCDAT_PROMPT_STRING @UVCDAT_PROMPT_STRING@
+if ( $?UVCDAT_PROMPT_BEGINING ) then
```
Comment (dlonie) [inline CMake/cdat_modules_extra/setup_runtime.csh.in]: Should be "BEGINNING"

File: CMake/cdat_modules_extra/setup_runtime.sh.in [hunk 2]
```diff
@@ -35,8 +35,12 @@ library_paths=( @SETUP_LIBRARY_PATHS@ )
 python_paths=( @SETUP_PYTHON_PATHS@ )
 executable_paths=( @SETUP_EXECUTABLE_PATHS@ )
 
-if [ -z "$UVCDAT_DISABLE_PROMPT" ] ; then
-    export PS1="(uvcdat)$PS1"
+export UVCDAT_PROMPT_STRING=@UVCDAT_PROMPT_STRING@
+if [ "$UVCDAT_ENABLE_PROMPT_BEGINING" ] ; then
```
Comment (dlonie) [inline CMake/cdat_modules_extra/setup_runtime.sh.in]: Same

**Comment by aashish24** [comment]
@doutriaux1 if you can fix the spelling, then I am okay with the changes.

**Comment by doutriaux1** [comment]
oh you know I only speak computer :wink:

**Comment by doutriaux1** [comment]
@aashish24 @dlonie I think it's in english now. Double check and approve if you want please.

**Comment by doutriaux1** [comment]
actually multiple sourcing will probably lead to many [uvcdat] in prompt. Still a bad approach, maybe we should just stick to setting ${UVCDAT_PROMPT_STRING}

**Comment by aashish24** [comment]
:laughing: LGTM

**Pull request merged**